.github/workflows/ci.yml

```diff
@@ -37,12 +37,14 @@ jobs:
       matrix:
         include:
           - # If upgrading the Haskell image, also upgrade it in the lint job below
-            os: "ubuntu-latest"
+            os: ["ubuntu-latest"]
             image: haskell:9.2.5@sha256:2597b0e2458165a6635906204f7fac43c22e7d2a46aca1235a811194bb6cd419
-          - os: "macOS-11"
-          - os: "windows-2019"
+          - os: ["macOS-11"]
+          - os: ["windows-2019"]
+          - os: ["self-hosted", "macos", "ARM64"]
+          - os: ["self-hosted", "Linux", "ARM64"]
 
-    runs-on: "${{ matrix.os }}"
+    runs-on: ${{ matrix.os }}
```

Comment:
Unnecessary. It's easy to confuse the YAML outer language with the GitHub Actions inner language, and I was intentionally very pedantic with the quotation marks in part to help clarify this. The `${{ }}` interpolation is a feature of GitHub Actions and not YAML; in the YAML encoding, this is still a string, even if it refers to a list once GitHub Actions interprets it.

Reply:
I don't mind either version, will add quotes everywhere (and arguably that should be the preferred way of doing things, since it's just safer)